### PR TITLE
Include inflation in monthly amortization table

### PR DESCRIPTION
**Pull request by hdavidzhu**
image

### CODE REVIEW

File: src/components/getAnnualResultsByOption.js [hunk 1]
```diff
@@ -96,24 +81,73 @@ export function getAnnualResultsByOption ({
...
+  const resultAfterInflation = {}
+
+  _.forOwn(
+    resultBeforeInflation,
+    (value, key) =>
+      (resultAfterInflation[key] = getAnnualValueAfterInflation(
+        value,
+        inflationRate
+      ))
+  )
```
Comment (hdavidzhu) [inline src/components/getAnnualResultsByOption.js]: @jgozal I don't think this math is actually right. It's not matching up what the other guy had. Will need to dig further, b/c I'm struggling to get inflation to work correctly w/ each time step

Comment (jgozal) [reply src/components/getAnnualResultsByOption.js]: no worries @hdavidzhu! it's definitely not easy

Comment (jgozal) [reply src/components/getAnnualResultsByOption.js]: no worries @hdavidzhu  I'm sure it's not as easy as we thought

Comment (jgozal) [reply src/components/getAnnualResultsByOption.js]: @hdavidzhu please let me know when this PR is ready for review.

Comment (hdavidzhu) [reply src/components/getAnnualResultsByOption.js]: Go for it!

Comment (jgozal) [reply src/components/getAnnualResultsByOption.js]: nice!! I love how it looks, will review the code now

File: src/components/main-form.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ import numeral from 'numeral';
 import styled, { css } from "react-emotion";
 import { Input } from 'reactstrap';
 import { FV, PMT } from 'formulajs/lib/financial';
-import { getAnnualResultsByOption } from './getAnnualResultsByOption'
+import { getMonthlyResultsByOption } from './getMonthlyResultsByOption'
```
Comment (jgozal) [inline src/components/main-form.js]: I'm not used to seeing files named as functions, is that a thing?  If so we should probably move it out of the `components` folder and put it in a `utils` folder or something along those lines. I would even name the file something like `mortgage-utils.js` and add in more useful functions in there but don't mind if you prefer to name it like the actual function as long as we keep it consistent

Comment (hdavidzhu) [reply src/components/main-form.js]: Yeah I think the file name is ok for now, and if want to put more functions in the same file, then we can rename in the future. I'll move it out tho

File: src/components/getMonthlyResultsByOption.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+function getMonthlyResult (
+  mortgageRate,
+  investmentRate,
+  inflationRate,
+  lastResult
+) {
+  let pmt = lastResult.pmt
+
+  if (
+    lastResult.loanAmount < FV(mortgageRate / COMPOUND_FREQUENCY, 1, pmt, 0)
```
Comment (jgozal) [inline src/components/getMonthlyResultsByOption.js]: could you document a little of the thought process here?

File: src/components/main-form.js [hunk 4]
```diff
@@ -209,7 +213,34 @@ class MainForm extends React.Component {
             onChange={this.updateInput}
           />
         </div>
-
+        <div className='row'>
+          <div className='col font-weight-bold'>Month</div>
+          <div className='col font-weight-bold'>{this.state.shorterOption.term} Yr Mortgage Payment</div>
+          <div className='col font-weight-bold'>{this.state.shorterOption.term} Yr Investment Payment</div>
```
Comment (jgozal) [inline src/components/main-form.js]: You could really use reacstrap directly here and make this a lot cleaner and explicit: https://reactstrap.github.io/components/layout/

For the table's header, I would create a styled component named `ColHeader` like so:

const ColHeader = styled(Col)`
  font-weight: bold;
`

Less CSS floating around, more composition, and much more explicit (less divs and more Cols, ColHeaders and Rows)

File: src/utils/getMonthlyResultsByOption.js [hunk 5]
```diff
@@ -91,12 +91,15 @@ function getMonthlyResult (
 ) {
   let pmt = lastResult.pmt
 
+  // If the loan is less than the monthly payment, only pay what's remaining in
+  // the loan amount
```
Comment (jgozal) [inline src/utils/getMonthlyResultsByOption.js]: thank you!

File: src/components/main-form.js (hunk 4 above)
Comment (hdavidzhu) [reply src/components/main-form.js]: Where do I get `Col`? Is it possible to extend from Bootstrap to get `col`?

Comment (jgozal) [reply src/components/main-form.js]: See link in my first comment:

`import { Container, Row, Col } from 'reactstrap';`

We're already using reactstrap for our `Input` component. There is almost never a reason to use Bootstrap directly if we're using reactstrap.

File: src/components/main-form.js [hunk 6]
```diff
@@ -4,11 +4,12 @@
 // https://michaelbluejay.com/house/15vs30.html
 
 import _ from 'lodash';
-import React from 'react';
+import { Col, Row } from 'reactstrap'
+import { FV, PMT } from 'formulajs/lib/financial';
+import { Input } from 'reactstrap';
```
Comment (jgozal) [inline src/components/main-form.js]: double import of reactstrap, Other than that looks great. Thank you!

File: src/components/main-form.js [hunk 7]
```diff
@@ -209,7 +213,34 @@ class MainForm extends React.Component {
...
+            <Row>
+              <Col>{month}</Col>
+              <Col>{formatMoney(shorter.pmt)}</Col>
+              <Col>{formatMoney(shorter.investmentPMT)}</Col>
+              <Col>{formatMoney(shorter.loanAmount)}</Col>
+              <Col>{formatMoney(shorter.investmentAmount)}</Col>
+              <Col>{formatMoney(longer.pmt)}</Col>
+              <Col>{formatMoney(longer.investmentPMT)}</Col>
+              <Col>{formatMoney(longer.loanAmount)}</Col>
+              <Col>{formatMoney(longer.investmentAmount)}</Col>
```
Comment (jgozal) [inline src/components/main-form.js]: way easier to read now

**Pull request merged**